for try catch slect where -> try with aop to catch the exception